### PR TITLE
add TLS config

### PR DESCRIPTION
**Pull request by vixns**
```
clusters:
- name: test
  brokers:
  - localhost:9092
  SASL:
    mechanism: PLAIN
    username: admin
    password: mypasswordisnotsosimple
  TLS:
    cafile: /path/ca.pem
    verify: true
  security-protocol: SASL_SSL
```

### CODE REVIEW

File: cmd/kaf/kaf.go [hunk 1]
```diff
@@ -20,6 +22,23 @@ func getConfig() (saramaConfig *sarama.Config) {
...
+			caCertPool := x509.NewCertPool()
+			caCertPool.AppendCertsFromPEM(caCert)
+
+			tlsConfig := &tls.Config{
+				RootCAs:            caCertPool,
+				InsecureSkipVerify: cluster.TLS.Verify,
+			}
+			saramaConfig.Net.TLS.Config = tlsConfig
+		} else {
+			saramaConfig.Net.TLS.Config = &tls.Config{ InsecureSkipVerify:true }
```
Comment (birdayz) [inline cmd/kaf/kaf.go]: i think this changes the current behavior. currently, the system ca certs are used. in this case, this turns off security and all certs will be accepted, which is a security hole.
so this else block should be removed. we could add another config option for insecureSkipVerify, is a user really wants to do it

**Comment by birdayz** [comment]
we should really have some sample configs, because currently the kaf config add-cluster command does not cover all cases, also here. and i suspect that it will be difficult getting all config options into this command.
so, can you add your example config into a file in a new folder examples ? and refer to it from the reference? that would be really nice!

File: cmd/kaf/kaf.go (hunk 1 above)
Comment (vixns) [reply cmd/kaf/kaf.go]: fixed in 1fd8a80

**Comment by vixns** [comment]
updated

**Comment by birdayz** [comment]
very nice. thanks!

**Comment by fuyar** [comment]
You can't use client SSL files for auth right now am i right ?

**Comment by vixns** [comment]
@fuyar you're right, this is not implemented for now.

**Pull request merged**